packages/@react-aria/landmark/test/useLandmark.test.tsx

```diff
@@ -0,0 +1,866 @@
+    expect(document.activeElement).toBe(navigation);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(main);
+
+    tree.rerender(
+      <Container>
+        <Region>
+          <TextField label="First Name" />
+        </Region>
+      </Container>
+    );
+
+    let region = tree.getByRole('region');
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(region);
+  });
+
+  it('Should navigate to a landmark that has been added as a child to an existing landmark.', function () {
+
+    function Container({children = null}) {
+      return (
+        <div>
+          <Navigation>
+            <ul>
+              <li><a href="/home">Home</a></li>
+              <li><a href="/about">About</a></li>
+              <li><a href="/contact">Contact</a></li>
+            </ul>
+          </Navigation>
+          <Main>
+            <TextField label="Last Name" />
+            {children}
+          </Main>
+        </div>
+      );
+    }
+    let tree = render(<Container />);
+
+    tree.rerender(
+      <Container>
+        <Region>
+          <TextField label="First Name" />
+        </Region>
+      </Container>
+    );
+    let main = tree.getByRole('main');
+    let navigation = tree.getByRole('navigation');
+    let region = tree.getByRole('region');
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(navigation);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(main);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(region);
+  });
+
+  it('Should navigate to a landmark that has been added as a parent to an existing landmark.', function () {
+
+    function Contained() {
+      return (
+        <div>
+          <Navigation>
+            <ul>
+              <li><a href="/home">Home</a></li>
+              <li><a href="/about">About</a></li>
+              <li><a href="/contact">Contact</a></li>
+            </ul>
+          </Navigation>
+          <Region>
+            <TextField label="Last Name" />
+          </Region>
+        </div>
+      );
+    }
+
+    let tree = render(<Contained />);
+
+    tree.rerender(
+      <Main>
+        <TextField label="Last Name" />
+        <Contained />
+      </Main>
+    );
+
+    let main = tree.getByRole('main');
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(main);
+  });
+
+  it('can alt+F6 to main landmark', function () {
```

Comment:
lets also add a test for no Main landmark, as well as one with only a Main and no other landmarks